packages/siop-oid4vp/lib/authorization-response/AuthorizationResponse.ts

```diff
@@ -124,21 +127,31 @@ export class AuthorizationResponse {
     })
 
     if (hasVpToken) {
-      const wrappedPresentations = response.payload.vp_token
-        ? await extractPresentationsFromVpToken(response.payload.vp_token, {
+      if (responseOpts.presentationExchange) {
+        const wrappedPresentations = response.payload.vp_token
+          ? await extractPresentationsFromVpToken(response.payload.vp_token, {
+              hasher: verifyOpts.hasher,
+            })
+          : []
+
+        await assertValidVerifiablePresentations({
+          presentationDefinitions,
+          presentations: wrappedPresentations,
+          verificationCallback: verifyOpts.verification.presentationVerificationCallback,
+          opts: {
+            ...responseOpts.presentationExchange,
             hasher: verifyOpts.hasher,
-          })
-        : []
-
-      await assertValidVerifiablePresentations({
-        presentationDefinitions,
-        presentations: wrappedPresentations,
-        verificationCallback: verifyOpts.verification.presentationVerificationCallback,
-        opts: {
-          ...responseOpts.presentationExchange,
+          },
+        })
+      } else {
+        const dcqlQuery = verifiedAuthorizationRequest.dcqlQuery
+        if (!dcqlQuery) {
+          throw new Error('vp_token is present, but no presentation definitions or dcql query provided')
+        }
+        assertValidDcqlPresentationRecord(responseOpts.dcqlQuery.encodedPresentationRecord as DcqlPresentationRecord, dcqlQuery, {
           hasher: verifyOpts.hasher,
-        },
-      })
+        })
+      }
```

Comment:
nit but to make this easier to read / have the same checks on the same level I'd do it as:

```
if (pd) {

}
else if (dcql) {

}
else {
	throw error
}
```